solr/modules/monitor/src/java/org/apache/lucene/monitor/MonitorFields.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *
+ *  * Licensed to the Apache Software Foundation (ASF) under one or more
+ *  * contributor license agreements.  See the NOTICE file distributed with
+ *  * this work for additional information regarding copyright ownership.
+ *  * The ASF licenses this file to You under the Apache License, Version 2.0
+ *  * (the "License"); you may not use this file except in compliance with
+ *  * the License.  You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.apache.lucene.monitor;
+
+import java.util.Set;
+
+public class MonitorFields {
+
+  public static final String QUERY_ID = QueryIndex.FIELDS.query_id + "_";
+  public static final String CACHE_ID = QueryIndex.FIELDS.cache_id + "_";
+  public static final String MONITOR_QUERY = QueryIndex.FIELDS.mq + "_";
+  public static final String PAYLOAD = QueryIndex.FIELDS.mq + "_payload_";
```

Comment:
The https://github.com/cpoerschke/solr/commit/26f932bc077cc2fc1abd553480e9efe715a6f7c5 commit on https://github.com/cpoerschke/solr/commits/solr-monitor-cpoerschke-3 explores turning the `MonitorFields.PAYLOAD` constant into a configurable.

And interesting insight from that is the co-occurrence of `org.apache.lucene.monitor.QueryDecomposer` and `String payloadFieldName` -- not yet explored if maybe inheritance might be a thing.

```
class SolrQueryDecomposer extends org.apache.lucene.monitor.QueryDecomposer {
  String payloadFieldName;
}
```

Also (note-to-self) looking to understand further w.r.t. query decomposer vs. query decoder and how the field names fit into that picture.

Reply:
So echoing one of your earlier points about limiting the scope of the initial phase, payload is one of those things that would be on my cutting block. Initially I tried to have something that would effectively equate to the [optional lucene-monitor metadata map](https://github.com/apache/lucene/blob/main/lucene/monitor/src/java/org/apache/lucene/monitor/MonitorQuery.java#L43). I believe the point of this is to return back that metadata in case there is something useful that you want to associate with an indexed query. Functionally, it serves the same purpose to a stored/doc-values solr field but in a much less flexible and less normalized way. What's funny is that I haven't actually wired this into the response yet because I don't think any of the default lucene-monitor responses pass it back and I was simply going for feature parity. That aside, is there a particular reason you were exploring making the payload field name overridable? Maybe to let users give it a more self-describing name for their particular use case?

The difference between a query decomposer and a decoder is that the decomposer takes an in-memory `Query` object and splits it into `List<Query>` where each element is a presumably smaller disjunct that is less expensive to match in the final phase (think for example of an or-expression with 500 terms .. each one of those can be indexed/matched separately for a much faster matching phase). A query decoder simply takes the original full query string and parses it to the original full/undecomposed `Query`. After that you have to split it into the disjuncts again as [there is no _easy_ and _general_ way](https://stackoverflow.com/a/16966866) to serialize those inner `Query`s. That detail aside, I don't see any reason the decoder can't use the decomposer directly and expose that decomposition as part of its API ... this is more of me explaining this distinction between lucene's original `MonitorQuerySerializer` and `QueryDecomposer`.

Reply:
> ... is there a particular reason you were exploring making the payload field name overridable? Maybe to let users give it a more self-describing name for their particular use case? ...

So I was exploring not specifically w.r.t. making the payload field name overridable but the other fields (query id, cache id, query) also and payload was just first to explore. Yes, optionally-choosable names would let users choose names/terminology to suit their use case, and perhaps might it also then reduce or remove the need for the aliasing functionality (which I haven't yet looked into further)?

As an aside, I noticed that the query id and cache id and query field name choices picked the Lucene `QueryIndex` fields -- https://github.com/apache/lucene/blob/releases/lucene/9.10.0/lucene/monitor/src/java/org/apache/lucene/monitor/QueryIndex.java#L46-L48 -- and suffixed an `_` underscore, is the suffixing needed or does it just make maybe debugging or so easier?

Reply:
> So I was exploring not specifically w.r.t. making the payload field name overridable but the other fields (query id, cache id, query) also and payload was just first to explore. Yes, optionally-choosable names would let users choose names/terminology to suit their use case, and perhaps might it also then reduce or remove the need for the aliasing functionality (which I haven't yet looked into further)?

Ok I am on board with the configuration then. I don't necessarily think it can obviate aliasing because aliasing only applies to fields dynamically generated by the presearcher. These don't exist in `MonitorFields`. For instance, say you have a query `foo:bar` and you naturally have some schema definition that `foo` conforms to in your collection of _regular_ documents. You now want to create a collection of queries for reverse search and we want to make that as easy as possible. So you use should be able to use the same schema (with the same query analyzers etc) with the addition of a few control fields. One of those is `__....monitor_alias_*` so that when the presearcher is done tokenizing that query `foo:bar` it actually sends the token stream to the field `__....monitor_alias_foo`. So if your original `foo` field had all sorts of fancy configurations like `docValues` or `stored`, the presearcher generated field won't know about any of that and will instead give you a plain old indexed field. Now because of what appears to be an implementation choice, solr will actually let you write a field with a configuration that is different from the schema because the only validation is against the first _written_ field (to make sure that there is compatibility between the first document's field value and all the documents that come after). But relying on this feels hacky. Also, the schema discrepancy mentioned above would be a total blocker if you wanted to store queries alongside your documents in the same collection (because the first written document would be a "true" document with whatever fancy configuration you had defined in the schema). Finally, the multi-pass presearcher does things to field names that also warrant a more dynamic approach to handling it. Now if there was ever a presearcher that gave you back a different _type_ of field then we'd probably have to define another dynamic alias field .. but that doesn't seem like something that will happen very often.

> https://github.com/apache/lucene/blob/releases/lucene/9.10.0/lucene/monitor/src/java/org/apache/lucene/monitor/QueryIndex.java#L46-L48 -- and suffixed an _ underscore, is the suffixing needed or does it just make maybe debugging or so easier?

I kind of just assumed based on `_root_` and `_version_` that `_` wrapped fields are considered out of the user space, at least by convention, so I wanted to be consistent.

Reply:
something like https://github.com/kotman12/solr/pull/5 could make it configurable.